WalletWasabi/WabiSabi/Backend/PostRequests/InputRegistrationHandler.cs

```diff
@@ -127,10 +127,15 @@ public static class InputRegistrationHandler
 				throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.WrongPhase);
 			}
 
+			if (!round.TryAddMoreWeight((uint)inputWeightSum))
+			{
+				throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.TooMuchTotalWeight);
+			}
+
 			var amountCredentialResponse = round.AmountCredentialIssuer.HandleRequest(zeroAmountCredentialRequests);
 			var weightCredentialResponse = round.WeightCredentialIssuer.HandleRequest(zeroWeightCredentialRequests);
 
-			RemoveDuplicateAlices(rounds, alice);
+			RemoveDuplicateAlices(rounds, alice); //TODO: handle removed Alices weight credentials.
```

Comment:
This is addressed already, because duplicate inputs are only removed if they are still input registration, at which point only zero credentials have been issued (of both kinds).

https://github.com/zkSNACKs/WalletWasabi/blob/0446d4154ca107a8f91ed14a394cdb46f04718bc/WalletWasabi/WabiSabi/Backend/PostRequests/InputRegistrationHandler.cs#L150-L152